### PR TITLE
login refactored

### PR DESCRIPTION
**Pull request by suomiy**
login refactored into other restClients
sync refactored
new version of Android Annotations and spring-android-rest-template

---
This change is [Reviewable](https://reviewable.io/reviews/matobet/movirt/190)

### CODE REVIEW

File: moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java [hunk 1]
```diff
@@ -703,35 +726,35 @@ private void updateClientBeforeCall() {
...
         String token = "";
 
-        resetClientSettings();
-        setupVersionHeader("");
-        updateClientBeforeCall();
-        restClient.setRootUrl(authenticator.getBaseUrl());
-        try {
-            token = restClient.loginV4(username, password).getAccessToken();
-        } catch (Exception x) {// 405 Method Not Allowed - old API
+        synchronized (loginV4Lock) {
```
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java]: Not sure if we really need synchronization here. One big downside is, if the user picks wrong ip address (in valid format), login is locked until timeout occurs (which can be for quite a long time).

**Comment by matobet** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java, line 73 \[r1\]](https://reviewable.io:443/reviews/matobet/movirt/190#-KNiUzJ-ft-zaLW1Ua_c:-KNiUzJ-ft-zaLW1Ua_d:1693359614) ([raw file](https://github.com/matobet/movirt/blob/05d298658e7529bf170c7cdeb3220afaa7b7270c/moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java#L73)):*
> ```Java
> 
>     private static final String TAG = OVirtClient.class.getSimpleName();
>     private static final Object loginV3Lock = new Object();
> ```

those lock objects seem to be used only to synchronize access to loginV3 and V4 clients. I think it is better just to use the protected rest client objects themselves in syncrhonized statements. E.g.

```java
synchronized (loginV3RestClient) {
    ...
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/matobet/movirt/190)*

**Comment by suomiy** [comment]
InterruptedException still not fixed. Synchronized in SyncAdapter doesn't seem to help. But the exception occurs less. I can get it only in debug now. 

---

Review status: 0 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java, line 73 \[r1\]](https://reviewable.io:443/reviews/matobet/movirt/190#-KNiUzJ-ft-zaLW1Ua_c:-KNnFd2t3CxjhDs8o47i:-467915377) ([raw file](https://github.com/matobet/movirt/blob/05d298658e7529bf170c7cdeb3220afaa7b7270c/moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java#L73)):*
<details><summary><i>Previously, matobet (Martin Beták) wrote…</i></summary>
> those lock objects seem to be used only to synchronize access to loginV3 and V4 clients. I think it is better just to use the protected rest client objects themselves in syncrhonized statements. E.g.
> 
> ```java
> synchronized (loginV3RestClient) {
>     ...
> }
> ```
</details>
fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/matobet/movirt/190#-:-KNnIj8W2XJIaMJTdiym:-1946837097)*

**Comment by suomiy** [comment]
InterruptedException fixed in https://github.com/matobet/moVirt/pull/190/commits/a7523f0adec89fe2e387e542d8164c8989898041

**Pull request merged**